pkg/guid/guid_pool.go

```diff
@@ -50,6 +57,30 @@ func NewPool(conf *config.GUIDPoolConfig) (Pool, error) {
 	}, nil
 }
 
+func (p *guidPool) Reset(guids []string) error {
+	log.Debug().Msg("resetting guid pool")
+
+	p.guidPoolMap = map[GUID]bool{}
+	if guids == nil {
+		return nil
+	}
+
+	for _, guid := range guids {
+		err := p.AllocateGUID(guid)
+		if err != nil {
+			switch {
```

Comment:
alternative is to add a method: `guidInRange(guid) bool` and use it instead of swich case relying on error.

in fact most times you will have other guids configured in UFM.
my preference is to avoid relying on error of one method for a "good flow" of another.

that said, functionality wise it yields the same result so not a blocker imo

Reply:
Thanks, changed it.